### PR TITLE
Prod envs now use openforcefield 0.8.4

### PR DESCRIPTION
**Pull request by dotsdl**
## Description
Version bump for the openforcefield toolkit 0.8.4rc1 -> 0.8.4.

Shouldn't be much different than 0.8.4rc1, so shouldn't be dramatic.

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by dotsdl** [comment]
Something about 0.8.4 appears to have broken the tests. Based on the differences between 0.8.4 and 0.8.4rc1, is it obvious what may be causing this @j-wags?

**Comment by dotsdl** [comment]
CI runs on `master` appear to work fine still: https://github.com/openforcefield/openff-benchmark/actions/runs/647641024

**Comment by j-wags** [comment]
Looking into this now. Apologies for the delay.

**Comment by j-wags** [comment]
Probably an interaction with [this change](https://github.com/openforcefield/openff-toolkit/pull/790/files#diff-68cb3f9873e00994ccf957cc089b18cbaf34302d5eeb90c34ca289803e10a69eR4663-R4672). Setting up a dev environment to resolve it.

**Comment by j-wags** [comment]
Ok, I found and fixed two problems:
* OFFTK 0.8.4 [pre-populates the mapped SMILES in the `extras` dict](https://github.com/openforcefield/openff-toolkit/pull/790/files#diff-68cb3f9873e00994ccf957cc089b18cbaf34302d5eeb90c34ca289803e10a69eR4663-R4672) when it runs `offmol.to_qcschema`. openff-benchmark then [tries to set it _again_](https://github.com/openforcefield/openff-benchmark/blob/44cb6b5f874859728193b7ef8b0bfce39d9e2a4d/openff/benchmark/geometry_optimizations/compute.py#L875) using the same mapped SMILES, which causes an error. So I added a try/except that should make it work for `0.8.4rc1` (which _doesn't_ pre-populate the mapped SMILES) AND 0.8.4 (which _does_ pre-populate it). This fix seems to work on my local machine. @dotsdl, could you sanity-check this?
* SQLAlchemy 1.4 came out a few hours ago and is causing tests to fail. I don't know much about our use of that package. So I've pinned to 1.3.

**Review by j-wags — APPROVED** [review]
I think this looks good, but per my message above, I'd love a second set of eyes!

File: openff/benchmark/geometry_optimizations/compute.py [hunk 1]
```diff
@@ -701,8 +709,15 @@ def _generate_optimization_input(mol, compute_spec, factory):
         # TODO: bug report in openff where `atom_map` is a string
         if isinstance(mol.properties.get('atom_map'), str):
             mol.properties['atom_map'] = ast.literal_eval(mol.properties['atom_map'])
-    
-        attributes = factory.create_cmiles_metadata(mol)
+
+        # This block will fail for OFF Toolkit 0.8.4, but succeed for 0.8.4rc1
+        try:
```
Comment (dotsdl) [inline openff/benchmark/geometry_optimizations/compute.py:714]: Good catch @j-wags!

**Comment by dotsdl** [comment]
Ah, thank you for this @j-wags! Reading your resolution it's clear I should have dove into it further myself, since I was directly involved in those changes.

**Review by dotsdl — COMMENTED** [review]
Thank you @j-wags! Good eye!

**Pull request merged**